web_src/js/features/notification.js

```diff
@@ -27,35 +45,108 @@ export function initNotificationCount() {
-    const source = new EventSource(`${AppSubUrl}/user/events`);
-    source.addEventListener('notification-count', async (e) => {
-      try {
-        const data = JSON.parse(e.data);
-
-        const notificationCount = $('.notification_count');
-        if (data.Count === 0) {
-          notificationCount.addClass('hidden');
-        } else {
-          notificationCount.removeClass('hidden');
-        }
 
-        notificationCount.text(`${data.Count}`);
-        await updateNotificationTable();
-      } catch (error) {
-        console.error(error);
-      }
-    });
-    source.addEventListener('logout', async (e) => {
-      if (e.data !== 'here') {
-        return;
-      }
-      source.close();
-      window.location.href = AppSubUrl;
-    });
-    window.addEventListener('beforeunload', () => {
-      source.close();
-    });
-    return;
+    if (window.SharedWorker && NotificationSettings.UseSharedWorker) {
+      // const {default: Worker} = await import(/* webpackChunkName: "eventsource" */'./eventsource.sharedworker.js');
+      // const worker = Worker('notification');
+      const worker = new SharedWorker('js/eventsource.sharedworker.js', 'notification-worker');
+      // worker.port.addEventListener('message', (event) => {
+      //   console.log(event.data);
+      // }, false);
+      // worker.port.start();
+      worker.addEventListener('error', (event) => {
+        console.error(event);
+      }, false);
+      worker.port.onmessageeerror = (event) => {
+        console.error(event);
+      };
+      worker.port.postMessage({
+        type: 'start',
+        url: `${window.location.protocol}//${window.location.host}${AppSubUrl}/user/events`,
+      });
+      worker.port.addEventListener('message', (e) => {
+        if (!e.data || !e.data.type) {
+          console.error(e);
+          return;
+        }
+        switch (event.data.type) {
+          case 'notification-count':
+            receiveUpdateCount(e.data);
+            return;
+          case 'error':
+            console.error(e.data);
+            return;
+          case 'logout': {
+            if (e.data !== 'here') {
+              return;
+            }
+            worker.port.postMessage({
+              type: 'close',
+            });
+            worker.port.close();
+            window.location.href = AppSubUrl;
+            return;
+          }
+          default:
+            return;
+        }
+      }, false);
+      worker.port.addEventListener('error', (e) => {
+        console.error(e);
+      }, false);
+      worker.port.start();
+      window.addEventListener('beforeunload', () => {
+        worker.port.postMessage({
+          type: 'close',
+        });
+        worker.port.close();
+      }, false);
+
+      return;
+    }
+
+    if (window.Worker && NotificationSettings.UseWorker) {
+      const {default: Worker} = await import(/* webpackChunkName: "eventsource" */'./eventsource.worker.js');
```

Comment:
What's the benefit of using a plain worker? Is it not restricted by that 6-connection limit?

Reply:
Happy to remove - I was just including it for completeness to be honest.

Reply:
Removed